src/modules/accommodations/controllers/accommodations.controller.ts

```diff
@@ -1,32 +1,61 @@
+import { AccommodationAmenityService } from '#/modules/accommodations/services/accommodation-amenity.service';
+import { AccommodationAddressDto } from '#/modules/accommodations/dto/requests/create-accommodation-address.req';
+import { AccommodationImageDto } from '#/modules/accommodations/dto/requests/create-accommodation-image.req';
+import { AccommodationAmenityDto } from '#/modules/accommodations/dto/requests/create-accommodation-amenity.req';
+import { AccommodationAddress } from '#/modules/accommodations/entities/accommodation-address.entity';
+import { AccommodationImage } from '#/modules/accommodations/entities/accommodation-image.entity';
+import { AccommodationAmenity } from '#/modules/accommodations/entities/accommodation-amenity.entity';
+import { Accommodation } from '#/modules/accommodations/entities/accommodations.entity';
+import { AccommodationDto } from '#/modules/accommodations/dto/requests/create-accommodation.req';
 
-@Controller('accommodation')
-export class AccommodationsController {
-  constructor(private readonly AccommodationService: AccommodationsService) {}
+@Controller('accommodations')
+export class AccommodationController {
+  constructor(
+    private readonly accommodationService: AccommodationService,
+    private readonly accommodationAddressService: AccommodationAddressService,
+    private readonly accommodationImageService: AccommodationImageService,
+    private readonly accommodationAmenityService: AccommodationAmenityService,
+  ) {}
 
   @Post()
-  create() {
-    return this.AccommodationService.create();
+  async createAccommodation(
+    @Body() createAccommodationDto: AccommodationDto,
+  ): Promise<Accommodation> {
+    return this.accommodationService.create(createAccommodationDto);
   }
 
-  @Get()
-  findAll() {
-    return this.AccommodationService.findAll();
+  // Route to create an address for an accommodation
+  @Post(':accommodationId/address')
+  async createAddress(
+    @Param('accommodationId') accommodationId: string,
+    @Body() addressDto: AccommodationAddressDto,
+  ): Promise<AccommodationAddress> {
+    return this.accommodationAddressService.create(accommodationId, addressDto);
   }
 
-  @Get()
-  findOne() {
-    return this.AccommodationService.findOne();
+  // Route to add images to an accommodation
+  @Post(':accommodationId/images')
+  async addImages(
+    @Param('accommodationId') accommodationId: string,
+    @Body() imagesDto: AccommodationImageDto[],
+  ): Promise<AccommodationImage[]> {
+    return Promise.all(
+      imagesDto.map((imageDto) =>
+        this.accommodationImageService.create(accommodationId, imageDto),
+      ),
+    );
   }
 
-  @Put()
-  update() {
-    return this.AccommodationService.update();
-  }
-
-  @Delete()
-  remove() {
-    return this.AccommodationService.remove();
+  // Route to add amenities to an accommodation
+  @Post(':accommodationId/amenity')
+  async createAmenity(
```

Comment:
Not sure that we need this endpoint. Amenities are created when accommodation is created